### PR TITLE
Add missing roles to tls environment file

### PR DESCRIPTION
**Pull request by olliewalsh**
Adding roles for subnetted deployments to tlse environment.

### CODE REVIEW

File: ansible/templates/osp/tripleo_heat_envs/common/tlse.yaml.j2 [hunk 1]
```diff
@@ -16,10 +17,12 @@ resource_registry:
   OS::TripleO::Services::HAProxyInternalTLS: deployment/haproxy/haproxy-internal-tls-certmonger.yaml
   OS::TripleO::Services::IpaClient: deployment/ipa/ipaservices-baremetal-ansible.yaml
   OS::TripleO::Services::TLSProxyBase: deployment/apache/apache-baremetal-puppet.yaml
-  OS::TripleO::ControllerServiceServerMetadataHook: extraconfig/nova_metadata/krb-service-principals/controller-role.yaml
-  OS::TripleO::Compute{% if "hci" in osp.extrafeatures %}HCI{% endif %}ServiceServerMetadataHook: extraconfig/nova_metadata/krb-service-principals/compute{% if "hci" in osp.extrafeatures %}hci{% endif %}-role.yaml
+  {%- for role in ["Controller", "Compute", "ComputeLeaf1", "ComputeLeaf2", "ComputeHCI", "ComputeHCILeaf1", "ComputeHCILeaf2"] %}
+  OS::TripleO::{{ role }}ServiceServerMetadataHook: extraconfig/nova_metadata/krb-service-principals/{{ role|lower }}-role.yaml
+  {%- endfor %}
```
Comment (stuggi) [inline ansible/templates/osp/tripleo_heat_envs/common/tlse.yaml.j2:22]: How about looping through the osp.vmset and osp.bmset to not have to hard code the list here? I know we still have them in the roles file, but one less file with hard coded information.

```
  {%- for role in osp.vmset %}
  OS::TripleO::{{ role }}ServiceServerMetadataHook: extraconfig/nova_metadata/krb-service-principals/{{ role|lower }}-role.yaml
  {%- endfor %}
  {%- for role in osp.bmset %}
  OS::TripleO::{{ role }}ServiceServerMetadataHook: extraconfig/nova_metadata/krb-service-principals/{{ role|lower }}-role.yaml
  {%- endfor %}
```

Comment (stuggi) [reply ansible/templates/osp/tripleo_heat_envs/common/tlse.yaml.j2:22]: no this won't work if we have all the roles in the roles file. we might want to look into creating the roles file for the release and specified roles using `openstack overcloud roles generate`

Comment (olliewalsh) [reply ansible/templates/osp/tripleo_heat_envs/common/tlse.yaml.j2:22]: I didn't think it was worth it as this file is just temporary until I switch to using the .j2 environment file

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**